tpm/caps.go

```diff
@@ -0,0 +1,73 @@
+package tpm
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/google/go-tpm/legacy/tpm2"
+	"go.step.sm/crypto/tpm/algorithm"
+)
+
+type Capabilities struct {
+	Algorithms []algorithm.Algorithm
+}
+
+func (c *Capabilities) SupportsAlgorithms(algs ...tpm2.Algorithm) bool {
```

Comment:
This will require a user of the package to also import `github.com/google/go-tpm/legacy/tpm2` explicitly. Better to prevent that by using our own type or alias, so that it's easier to guarantee the public API. I believe `algorithm.Algorithm` would already be suitable for it.

Reply:
657022a